config/mediawiki/LocalSettings.php

```diff
@@ -8,7 +8,11 @@
 $wgAssumeProxiesUseDefaultProtocolPorts = false;
 
 ## Docker stuff
-if ( PHP_SAPI === 'cli' && !defined( 'MW_DB' ) ) {
+if (
+	( PHP_SAPI === 'cli' && !defined( 'MW_DB' ) )
+	||
+	( array_key_exists( 'SERVER_NAME', $_SERVER ) && $_SERVER['SERVER_NAME'] === 'web' )
```

Comment:
This doesn't check `!defined( 'MW_DB' )`.

If we want to remove support for setting this before this point (e.g. people that have their own LocalSttings.php file and load this one form their and have code above it), then I suggest removing it from the first part as well instead of sometimes working and sometimes not.

Personally, I think that's probably fine to remove indeed. I myself do use LocalSettings.php in this way, but I have all my code after the include, not above it.

Eg. this would become `PHP_SAPI === 'cli' || @$_SERVER['SERVER_NAME'] === 'web' `

Reply:
At least for the way browser tests work in the branch, the `( array_key_exists( 'SERVER_NAME', $_SERVER ) && $_SERVER['SERVER_NAME'] === 'web' )` would no longer be needed.
Would this affect how you run browser tests within fresh though?

Reply:
No, I run them through the public URL. Fine before/after.